step-templates/import-databricks-workbooks.json

```diff
@@ -0,0 +1,76 @@
\"PYTHON\" \n                $workbookFormat   = \"JUPYTER\"\n                break\n            }\n            '.scala' {\n                $workbookLanguage = \"SCALA\"\n                $workbookFormat   = \"SOURCE\"\n                break\n        }\n        Default \n            {\n                $workbookLanguage = \"SQL\"\n                $workbookFormat   = \"SOURCE\"\n            }\n        }\n        $requestBody = ConvertTo-Json -InputObject @{ \n            content = $workbookContent\n            path = $workbookPath\n            language = $workbookLanguage\n            format = $workbookFormat\n            overwrite = $true\n            }\n        \n        $apiResponse = Invoke-RestMethod -Method Post  -Uri  $Uri -Headers $headers -Body $requestBody\n        return $apiResponse\n    }\n}\n\n\nfunction Set-DatabricksWorkspaceFolder\n{\n    [CmdletBinding()]\n    param (\n        [Parameter()]\n        [String]\n        $AccessToken, \n        [Parameter()]\n        [String]\n        $DataBricksInstanceUri,\n        [Parameter()]\n        [String]\n        $DatabrickFolder\n    )\n\n    $headers = @{\n        'Authorization' = (\"Bearer {0}\" -f $AccessToken )\n    }\n    $APIVersion = '/api/2.0'\n    $APIListCommand = '/workspace/list'\n    $APIMkdirsCommand = '/workspace/mkdirs'\n    $ListUri = \"https://$DataBricksInstanceUri$APIVersion$APIListCommand\"\n    $MkdirsUri = \"https://$DataBricksInstanceUri$APIVersion$APIMkdirsCommand\"\n\n    $pathRoute = $DatabrickFolder.Substring(1) -split '/'\n    $basePath = \"/\"\n    foreach($path in $pathRoute)\n    {\n        $requestBody = @{ \n            path = $basePath\n        }\n        $apiResponse = Invoke-RestMethod -Uri $ListUri -Headers $headers -Body $requestBody -ContentType application/json  \n        $workSpaceFolder = $apiResponse.objects | Where-Object {$_.object_type -eq \"DIRECTORY\" -and $_.path -eq ( \"{0}{1}\" -f $basePath , $path) } \n        if($null -eq $workSpaceFolder)\n        {\n            $requestBody = ConvertTo-Json -InputObject  @{ \n                path = ( \"{0}{1}\" -f $basePath , $path)\n            } \n            Invoke-RestMethod -Method Post -Uri $MkdirsUri -Headers $headers -Body $requestBody\n        }\n        $basePath = \"{0}/{1}/\" -f $basePath , $path\n        if($basePath.StartsWith(\"//\"))\n        {\n            $basePath = $basePath.Substring(1)\n        }\n    }\n}\n\n#endregion fucntions\n\n\n$DatabrickWorkBookImportFolder = $OctopusParameters[\"Octopus.Action.Package[DeployDataBricksWorkBookPackage].ExtractedPath\"]\n\nWrite-Host \"Checking WorkSpace Folders\"\nSet-DatabricksWorkspaceFolder  -AccessToken $AccessToken -DataBricksInstanceUri $DataBricksInstanceUri -DatabrickFolder $DatabrickImportFolder\nWrite-Host \"Checking WorkSpace Folders\"\nSet-DatabricksWorkBook -AccessToken $AccessToken -DataBricksInstanceUri $DataBricksInstanceUri -WorkbooksUploadPath $DatabrickWorkBookImportFolder -DatabrickImportFolder $DatabrickImportFolder\n"
```

Comment:
On the script itself:
- There are 2 x `Write-Host "Checking WorkSpace Folders"` - should one of these be different?